extendr-api/src/error.rs

```diff
@@ -56,17 +55,87 @@ pub enum Error {
     ExpectedWeakRef(Robj),
     ExpectedRaw(Robj),
     ExpectedS4(Robj),
+    ExpectedPrimitive(Robj),
+
+    ExpectedScalar(Robj),
+    ExpectedVector(Robj),
+    ExpectedMatrix(Robj),
+    ExpectedMatrix3D(Robj),
+    ExpectedNumeric(Robj),
+    OutOfRange(Robj),
+    MustNotBeNA(Robj),
+    ExpectedNonZeroLength(Robj),
+    TypeMismatch(Robj),
 
     Other(String),
 }
 
-pub type Result<T> = std::result::Result<T, Error>;
-
 impl std::fmt::Display for Error {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{:?}", self)
+        match self {
+            Error::Panic => write!(f, "Panic detected."),
```

Comment:
The error messages in this long list of errors are inconsistent in whether they do or don't end in a period. I think they should all be the same.